_specifications/specification-3-17.md

```diff
@@ -8895,6 +8915,197 @@ export interface Moniker {
+	 */
+	readonly viewPort: Range;
+
+	/**
+	 * Additional information about the context in which inline values were
+	 * requested.
+	 */
+	readonly context: InlineValuesContext;
+}
+```
+
+<div class="anchorHolder"><a href="#inlineValuesContext" name="inlineValuesContext" class="linkableAnchor"></a></div>
+
+```typescript
+interface InlineValuesContext {
+	/**
+	 * The document range where execution has stopped.
+	 * Typically the end position of the range denotes the line where the inline values are shown.
+	 */
+	readonly stoppedLocation: Range;
+}
+```
+
+_Response_:
+* result: `InlineValue[]` \| `null` defined as follows:
+
+<div class="anchorHolder"><a href="#inlineValue" name="inlineValue" class="linkableAnchor"></a></div>
+
+```typescript
+/**
+ * Inline value information can be provided by different means:
+ * - directly as a text value (class InlineValueText).
+ * - as a name to use for a variable lookup (class InlineValueVariableLookup)
+ * - as an evaluatable expression (class InlineValueEvaluatableExpression)
+ * The InlineValue types combines all inline value types into one type.
+ */
+export type InlineValue = InlineValueText | InlineValueVariableLookup | InlineValueEvaluatableExpression;
+
+/**
+ * Provide inline value as text.
+ */
+export class InlineValueText {
+	/**
+	 * The document range for which the inline value applies.
+	 */
+	readonly range: Range;
+
+	/**
+	 * The text of the inline value.
+	 */
+	readonly text: string;
+}
+
+/**
+ * Provide inline value through a variable lookup.
+ * If only a range is specified, the variable name will be extracted from the underlying document.
+ * An optional variable name can be used to override the extracted name.
+ */
+export class InlineValueVariableLookup {
+	/**
+	 * The document range for which the inline value applies.
+	 * The range is used to extract the variable name from the underlying document.
+	 */
+	readonly range: Range;
+
+	/**
+	 * If specified the name of the variable to look up.
+	 */
+	readonly variableName?: string;
+
+	/**
+	 * How to perform the lookup.
+	 */
+	readonly caseSensitiveLookup: boolean;
+}
+
+/**
+ * Provide an inline value through an expression evaluation.
+ * If only a range is specified, the expression will be extracted from the underlying document.
+ * An optional expression can be used to override the extracted expression.
+ */
+export class InlineValueEvaluatableExpression {
+	/**
+	 * The document range for which the inline value applies.
+	 * The range is used to extract the evaluatable expression from the underlying document.
+	 */
+	readonly range: Range;
+
+	/**
+	 * If specified the expression overrides the extracted expression.
+	 */
+	readonly expression?: string;
+}
+```
+* error: code and message set in case an exception happens during the inline values request.
+
+#### <a href="#inlineValues_refresh" name="inlineValues_refresh" class="anchor">Inline Values Refresh Request (:arrow_right_hook:)</a>
```

Comment:
Why a request rather than a notification? Does the server really need a response?